File: 0682-baseball-game/0682-baseball-game.c
int calPoints(char** operations, int operationsSize) {
    int st[1001];
    int top = -1;
    for(int i = 0;i<operationsSize;i++){
        char ch = operations[i][0];
        if(ch != 'C' && ch != 'D' && ch != '+'){
            st[++top] = atoi(operations[i]); // push
        } 
        else if(ch == 'C') // pop
        {
            top--;
        }
           else if(ch == 'D')
        {
            int d = st[top]*2;
            st[++top] = d; // push doubled value into the stack
    }
    else {
        int a = st[top] + st[top-1];
        st[++top] = a; //  push sum
    }

}
int sum =0;
for(int i = top;i>=0;i--) sum+= st[i];
return sum;
    
}

File: 0682-baseball-game/0682-baseball-game.java
class Solution {
    public int calPoints(String[] operations) { 
        Stack<Integer>scores = new Stack<>();
        for(int  i = 0 ; i < operations.length ; i++)
        {
            if(operations[i].equals("C"))
            {
                scores.pop();
            }
            else if(operations[i].equals("D"))
            {
                scores.push(scores.peek()*2);
            }
            else if(operations[i].equals("+"))
            {
                int size = scores.size();
                scores.push(scores.get(size-1) + scores.get(size-2));
            }
            else
            {
                scores.push(Integer.parseInt(operations[i]));
            }
        }
        int sum = 0 ;
        while(!scores.empty())
        {
            sum += scores.pop();
        }
        return sum;
    }
}